src/App.jsx

```diff
@@ -51,7 +51,12 @@ export function App() {
 		<Router>
 			<Routes>
 				<Route path="/" element={<Layout />}>
-					<Route index element={<Home />} />
+					<Route
+						index
+						element={
+							<Home handleListTokenState={setListToken} listId={listToken} />
```

Comment:
How do we feel about changing the `element` prop to something like this? It would avoid the flicker you get when you click the home link and get redirected to the list route.
listToken ? (
	<List data={data} />
) : (
	<Home handleListTokenState={setListToken} listId={listToken} />
)

Reply:
@jeremiahfallin Love this too! Devin just mentioned something like this as well above.  @Yaosaur would you be down to replace this if time allows?

Reply:
Yeah I think it works slightly better? My friend suggested this for the ternary so we wouldn't have to remember to change the props for <`List />` in its own route, we would just import the `Navigate` component from `react-router-dom`:
```listToken ? (
								<Navigate to="/list" />
							) : (
								<Home handleListTokenState={setListToken} listId={listToken} />
							)```

Reply:
@jeremiahfallin @mentalcaries So I ended up trying the conditional rendering with Jeremiah's code block above and when I have a list token and I try navigating to Home, its still causing a flicker. 

I also tried 
```
listToken ? <List data={data} /> : <Home handleListTokenState={setListToken} listId={listToken} /> 
```
The above doesn't cause the flicker, but the problem with that is 
![image](https://user-images.githubusercontent.com/99099243/232263316-d753bdf7-8a43-4ed9-b892-9924b7932b35.png)
So its loading the component we want, but not the route we want so the Navlink still captures "/".

I got a working solution, but it does require a router overhaul. They key is using createBrowserRouter and using a loader. This flickering issue is talked about here: https://github.com/remix-run/react-router/issues/8295
Essentially, it seems like Navigate is triggering a useEffect and the flicker is the side effect of some rendering. The loader prevents that but does not work with our current router set up.

Reply:
I ended up implementing the conditional render for now. I think this is the better method compared to using useEffect: [see here](https://www.toptal.com/react/react-router-tutorial).

I am thinking in the long term we probably wouldn't be showing the home link anyways so its not very likely users will notice it.